File: CrabS2.java

// CrabS2.java

public class CrabS2 {
    public String name = "Rune Crab";
    public String type = "Tank";
    public int hp = 500;
    public int quantity = 2;

    public int absorb(int damage) {
        return Math.max(0, damage - 50);
    }
}

File: crab_s_2.py

# crab_s_2.py

class CrabS2:
    def __init__(self):
        self.name = "Rune Crab"
        self.type = "Tank"
        self.hp = 500
        self.quantity = 2

    def absorb(self, damage):
        return max(0, damage - 50)  # absorbs 50 damage by default
